clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -371,8 +371,7 @@ private void doSend(ClientRequest clientRequest, boolean isInternalRequest, long
         } catch (UnsupportedVersionException e) {
             // If the version is not supported, skip sending the request over the wire.
             // Instead, simply add it to the local queue of aborted requests.
-            log.debug("Version mismatch when attempting to send {} to {}",
-                    clientRequest.toString(), clientRequest.destination(), e);
+            log.debug("Version mismatch when attempting to send {} to {}", clientRequest, clientRequest.destination(), e);
```

Comment:
It does seem a bit odd as `ClientRequest#toString` has the destination. Should we just remove `clientRequest.destination()` as it isn't really adding any additional value?